models/dns.go

```diff
@@ -322,6 +85,35 @@ func fixTxt(recs []*RecordConfig) {
 	}
 }
 
+// CheckDomainIntegrity performs sanity checks on a DomainConfig
+// and panics if problems are found.
+func (dc DomainConfig) CheckDomainIntegrity() {
```

Comment:
Nothing should ever panic. Return an error.

Reply:
Should more of the normalize package be moved into models or vice versa? There are some poorly defined package boundaries here.